job_runner/reporting/event.py

```diff
@@ -87,78 +98,58 @@ def _init_handler(self, msg):
-
+    def _job_handler(self, msg: Message):
+        job_name = msg.job.name()
+        logger.debug(f"Job: {job_name} is being handle")
+        job_data = {
+            "source": f"{self._step_path()}/job/{msg.job.index}",
+            "datacontenttype": "application/json",
+        }
         if isinstance(msg, Start):
-            self._dump_event(
-                CloudEvent(
-                    {
-                        "type": _FM_JOB_START,
-                        "source": job_path,
-                        "datacontenttype": "application/json",
-                    },
-                    {
-                        "stdout": str(Path(msg.job.std_out).resolve()),
-                        "stderr": str(Path(msg.job.std_err).resolve()),
-                    },
-                )
+            logger.debug(f"{job_name} job is starting")
+            self._dump_cloud_event(
+                attributes={"type": _FM_JOB_START, **job_data},
+                data={
+                    "stdout": str(Path(msg.job.std_out).resolve()),
+                    "stderr": str(Path(msg.job.std_err).resolve()),
+                },
             )
             if not msg.success():
-                self._dump_event(
-                    CloudEvent(
-                        {
-                            "type": _FM_JOB_FAILURE,
-                            "source": job_path,
-                            "datacontenttype": "application/json",
-                        },
-                        {
-                            "error_msg": msg.error_message,
-                        },
-                    )
+                logger.warning(f"{job_name} job failed to start successfully")
+                self._dump_cloud_event(
+                    attributes={"type": _FM_JOB_FAILURE, **job_data},
+                    data={
+                        "error_msg": msg.error_message,
+                    },
                 )
 
         elif isinstance(msg, Exited):
-            if msg.success():
-                self._dump_event(
-                    CloudEvent(
-                        {
-                            "type": _FM_JOB_SUCCESS,
-                            "source": job_path,
-                        },
-                        None,
-                    )
-                )
-            else:
-                self._dump_event(
-                    CloudEvent(
-                        {
-                            "type": _FM_JOB_FAILURE,
-                            "source": job_path,
-                            "datacontenttype": "application/json",
-                        },
-                        {
-                            "exit_code": msg.exit_code,
-                            "error_msg": msg.error_message,
-                        },
-                    )
-                )
-
-        elif isinstance(msg, Running):
-            self._dump_event(
-                CloudEvent(
+            logger.debug(f"{job_name} job exiting")
```

Comment:
It has _exited_ this point.
```suggestion
            logger.debug(f"{job_name} job exited")
```

Maybe add exit code? Error message?